src/static.jl

```diff
@@ -3,66 +3,39 @@ using Distributed: @spawn
 using Base.Iterators: flatten
 using Serialization: serialize, deserialize
 
-const dbpath = joinpath(@__DIR__, "..", "db", "usingdb")
-const lockpath = joinpath(@__DIR__, "..", "db", "usingdb.lock")
+include("utils.jl")
 
+const PROGRESS_ID = "docseeker_progress"
 DOCDBCACHE = DocObj[]
 
 function _createdocsdb()
-  isfile(lockpath) && return
-
-  open(lockpath, "w+") do io
-    println(io, "locked")
-  end
-
+  @info "Docs" progress=0 _id=PROGRESS_ID
+  PKGSDONE[] = 0
   try
-    isfile(dbpath) && rm(dbpath)
-
-    pkgs = keys(installed())
-
-    if isdefined(Main, :Juno)
-      @eval import Juno
-      Juno.isactive() && begin
-        Juno.progress() do id
-          for (i, pkg) in enumerate(pkgs)
-            @info "caching documentations of $(pkg) ..." progress = i / length(pkgs) _id = id
-            process = @spawn _createdocsdb(pkg)
-            wait(process)
-          end
-        end
-      end
-    else
-      for (i, pkg) in enumerate(pkgs)
-        @info "caching documentations of $(pkg) ..."
-        process = @spawn _createdocsdb(pkg)
-        wait(process)
-      end
-    end
+    pkgs = collect(keys(installed()))
+    pushfirst!(pkgs, "Base")
+    ondone = (i, el) -> progress_callback(i, el, pkgs)
+    run_queued(docdb_wrapper, pkgs, ondone = ondone)
   catch err
     @error err
   finally
-    rm(lockpath)
+    @info "" progress=1 _id=PROGRESS_ID
   end
 end
```

Comment:
Maybe it would be better here to add `global DOCDBCACHE; DOCDBCACHE = _loaddocsdb()`, since otherwise calling `createdocsdb` multiple times make no difference on appearance.

Reply:
Since every `searchdocs` call will call this https://github.com/pfitzseb/DocSeeker.jl/blob/50bc29648f250bc3d6b16040fab6e8eaf19c9dca/src/static.jl#L65 and once `DOCDBCACHE` is decided then it would return the same `DOCDBCACHE`